crates/noirc_evaluator/src/ssa/ir/instruction/call.rs

```diff
@@ -75,7 +75,9 @@ pub(super) fn simplify_call(
             if let Some(length) = dfg.try_get_array_length(arguments[0]) {
                 let length = FieldElement::from(length as u128);
                 SimplifyResult::SimplifiedTo(dfg.make_constant(length, Type::field()))
-            } else if matches!(dfg.type_of_value(arguments[1]), Type::Slice(_)) {
+            } else if arguments.len() > 1
+                && matches!(dfg.type_of_value(arguments[1]), Type::Slice(_))
```

Comment:
I assume the `arguments.len() > 1` was added because it was crashing previously. Do you know what cases it was crashing in? I would have thought `try_get_array_length` above would weed out all the array values.

Reply:
I added it because like you guessed it was crashing for me. I was also surprised as I thought the first case for `try_get_array_length` would catch all array values. After removing it I am no longer getting a crash and unable to reproduce my crash. Thus, I'm inclined to believe the crash was due to a broken implementation I had created elsewhere in the program while I was iterating on the code. 

I am going to remove the `arguments.len() > 1` for now so that this can be caught with future compiler changes.